usaspending_api/spending_explorer/v2/filters/explorer.py

```diff
@@ -97,17 +98,30 @@ def object_class(self):
     def recipient(self):
         # Recipients Queryset
         alt_set = (
-            self.alt_set.filter(~Q(transaction_obligated_amount=Decimal("NaN")), award__recipient__isnull=False)
+            self.alt_set.filter(~Q(transaction_obligated_amount=Decimal("NaN")))
+            .filter(
+                Q(award__latest_transaction__contract_data__awardee_or_recipient_legal__isnull=False)
+                | Q(award__latest_transaction__assistance_data__awardee_or_recipient_legal=False)
+            )
             .annotate(
-                id=F("award__recipient__recipient_name"),
+                id=Coalesce(
+                    "award__latest_transaction__contract_data__awardee_or_recipient_legal",
+                    "award__latest_transaction__assistance_data__awardee_or_recipient_legal",
+                ),
                 type=Value("recipient", output_field=CharField()),
-                name=F("award__recipient__recipient_name"),
-                code=F("award__recipient__recipient_name"),
+                name=Coalesce(
+                    "award__latest_transaction__contract_data__awardee_or_recipient_legal",
+                    "award__latest_transaction__assistance_data__awardee_or_recipient_legal",
+                ),
+                code=Coalesce(
+                    "award__latest_transaction__assistance_data__awardee_or_recipient_legal",
+                    "award__latest_transaction__contract_data__awardee_or_recipient_legal",
+                ),
             )
             .values("id", "type", "name", "code", "amount")
             .annotate(total=Sum("transaction_obligated_amount"))
             .order_by("-total")
-        )
+        ).distinct()
```

Comment:
Did you find that this distinct() was necessary?  It seems like it shouldn't be since you're summing.  Django should build a group by for you (and I have confirmed that it is) that effectively does the same thing.  So you're basically distincting twice.